staging_service/import_specifications/file_parser.py

```diff
@@ -166,14 +166,58 @@ def __post_init__(self):
 
 def parse_import_specifications(
     paths: tuple[Path],
-    file_type_resolver: Callable[[Path], FileTypeResolution]
+    file_type_resolver: Callable[[Path], FileTypeResolution],
+    log_error: Callable[[Exception], None]
 ) -> ParseResults:
     """
     Parse a set of import specification files and return the results.
 
     paths - the file paths to open.
     file_type_resolver - a callable that when given a file path, returns the type of the file or
         a parser for the file.
+    log_error - callable for logging an exception.
     """
+    if not paths:
+        return ParseResults(errors=tuple([Error(ErrorType.NO_FILES_PROVIDED)]))
+    # TODO check for None, whitespace only paths, '.', '..' & Path.resolve()
+    try:
+        return _parse(paths, file_type_resolver)
+    except Exception as e:
+        log_error(e)
+        errors = tuple([Error(ErrorType.OTHER, str(e))])
+        return ParseResults(errors=errors)
+
+def _parse(
+    paths: tuple[Path],
+    file_type_resolver: Callable[[Path], FileTypeResolution],
+) -> ParseResults:
     results = {}
     errors = []
+    for p in paths:
+        file_type = file_type_resolver(p)
+        if file_type.unsupported_type:
+            errors.append(Error(
+                ErrorType.PARSE_FAIL,
+                f"{file_type.unsupported_type} "
+                    + "is not a supported file type for import specifications",
+                SpecificationSource(p)
+            ))
+            continue
+        res = file_type.parser(p)
+        if res.errors:
+            errors.extend(res.errors)
+        else:
+            for data_type in res.results:
+                if data_type in results:
```

Comment:
Do we want to clear out the results if there is the multiple specs error, so we return only errors rather than a mix of errors and successes?

Reply:
Oh nevermind, if there is a mix, we just throw them away.